### PR TITLE
Allow for horizontal dendrogram 

### PR DESCRIPTION
**Pull request by oheil**
This allows for horizontal dendrograms, which is completely rotated 90 degree clockwise.

There is another (in my opinion not complete) solution to this in this PR:
https://github.com/JuliaPlots/StatsPlots.jl/pull/214
It is missing adjustments to the axes which is needed in some circumstances.
On the other hand, above PR is aimed for a better user experience with function clustermap.

I also added a complete example to the README.md for others to be able to create the heatmap with cluster dendrograms.
I had some problems to integrate the example image to the README.md so maybe this makes some effort on your side if you decide to merge this PR. Sorry but I didn't found the proper way to do this.

See also discussion
https://discourse.julialang.org/t/statsplots-dendrogram-on-the-right-side-or-rotated/31348

### CODE REVIEW

**Comment by oheil** [comment]
Something missing here?

File: src/dendrogram.jl [hunk 1]
```diff
@@ -1,4 +1,5 @@
-function treepositions(hc::Hclust, useheight::Bool)
+function treepositions(hc::Hclust, useheight::Bool, orientation=:v)
```
Comment (BeastyBlacksmith) [inline src/dendrogram.jl:1]: In Plots.jl we use the symbols `:vertical` and `:horizontal` for `orientation

File: src/dendrogram.jl [hunk 2]
```diff
@@ -17,22 +18,39 @@ function treepositions(hc::Hclust, useheight::Bool)
...
-    return xs, ys
+    if orientation == :h
+        return ys, xs
+    else
+        return xs, ys
+    end
 end
 
-@recipe function f(hc::Hclust; useheight=true)
+@recipe function f(hc::Hclust; useheight=true, orientation=:v )
```
Comment (BeastyBlacksmith) [inline src/dendrogram.jl:28]: how does this interact with the `orientation` keyword that already exists?

File: src/dendrogram.jl [hunk 3]
```diff
@@ -17,22 +18,39 @@ function treepositions(hc::Hclust, useheight::Bool)
...
-    treepositions(hc, useheight)
+    if orientation == :h
+        if useheight
+            hs=sum(hc.heights)
+            xlims --> (0, hs+hs*0.01)
+        else
+            xlims --> (0, Inf)
+        end
+        xshowaxis --> useheight
+        yforeground_color_axis := :white
```
Comment (BeastyBlacksmith) [inline src/dendrogram.jl:42]: IMO all `:=`s should be `-->`s in this recipe or is there an apparent reason that this should be non-overridable?

Comment (oheil) [reply src/dendrogram.jl]: Done as you said. Except for legend, as it was in the original code and it seems to make sense to disallow legends here.

File: src/dendrogram.jl (hunk 2 above)
Comment (oheil) [reply src/dendrogram.jl]: This I can't answer. I hope there is no side effect.

File: src/dendrogram.jl (hunk 1 above)
Comment (oheil) [reply src/dendrogram.jl]: Changed. Don't know anymore why I used :v and :h

**Comment by oheil** [comment]
conflict is no conflict anymore.

File: src/dendrogram.jl (hunk 3 above)
Comment (BeastyBlacksmith) [reply src/dendrogram.jl]: I'm fine with providing no legend by default, but I think if a user wants to have a legend, there should be a way to get one

Comment (oheil) [reply src/dendrogram.jl]: Of course. I have no opinion here, it is what the original author decided. Anyways, I changed it.

File: src/dendrogram.jl (hunk 2 above)
Comment (BeastyBlacksmith) [reply src/dendrogram.jl]: Does this flip the parabola or not?
```
plot([hc_row, (1:5).^2],yticks=false,xrotation=90,orientation=:horizontal)
```

Comment (oheil) [reply src/dendrogram.jl]: No, it's just not working:
```
julia> plot([hc_row, (1:5).^2],yticks=false,xrotation=90,orientation=:horizontal)
ERROR: Cannot convert Hclust{Float64} to series data for plotting
```
This behaviour isn't changed. This doesn't work in current StatsPlots neither.

Comment (BeastyBlacksmith) [reply src/dendrogram.jl]: hmm.. can't think of a good testcase here, I guess its fine. @dashw, you got objections?

Comment (oheil) [reply src/dendrogram.jl]: I think it's is very specific for dendrograms and should not radiate outsides.

**Comment by daschw** [comment]
Looks good to me, but there still seems to be an unresolved conflict.

**Comment by oheil** [comment]
The current master has
`xticks --> (1:nnodes(hc), string.(1:nnodes(hc))[hc.order])`
I changed the new version to reflect this, but the conflict remains.
Maybe it must be resolved manually?

**Comment by oheil** [comment]
The conflict lines shown are now in the `else `part. For the `if `part the lines changed to the y variants. 
So I would say, no conflict, but intended.

**Comment by daschw** [comment]
I resolved the conflict now. @oheil could you have a look if the current version has the correct changes?

**Comment by oheil** [comment]
Do you mean "master branch on oheil/StatsPlots.jl is correct ?" ? Yes, it is correct.

**Comment by oheil** [comment]
The file diffs look correct too.

**Comment by daschw** [comment]
Thanks, and thanks for your contribution!

**Pull request merged**